src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -4418,6 +4418,33 @@ module TcDeclarations =
             let isAtOriginalTyconDefn = true
             let core = MutRecDefnsPhase1DataForTycon(synTyconInfo, SynTypeDefnSimpleRepr.Exception r, implements1, false, false, isAtOriginalTyconDefn)
             core, extra_vals_Inherits_Abstractslots @ extraMembers
+            
+    let private CheckStaticAbstractSlots g (tcref: EntityRef) members =
+        let staticMembers =
+            [ for ttype in tcref.ImmediateInterfaceTypesOfFSharpTycon do
+                match stripTyEqnsAndMeasureEqns g ttype with
+                | TType_app(tyconRef = tcref) ->
+                    tcref.MembersOfFSharpTyconSorted
+                    |> List.filter(fun x -> not x.IsInstanceMember)
+                    |> List.map(fun x -> x.DisplayNameCore)
+                | _ -> () ]
+            |> List.concat
+                        
+        let implMembers =
+            [ for synMemberDef in members do
+                match synMemberDef with
+                | SynMemberDefn.Interface(members= Some(synMemberDefns)) ->
+                    [ for synMemberDefn in synMemberDefns do
+                        match synMemberDefn with
+                        | SynMemberDefn.Member(memberDefn = SynBinding(headPat = SynPat.LongIdent(longDotId = SynLongIdent(id = [ _; ident])))) -> ident
+                        | _ -> () ]
+                | _ -> ()
+            ]|> List.concat
+            
+        for memberId in implMembers do
+            if staticMembers |> List.exists(fun name -> name = memberId.idText)  then
```

Comment:
Is it safe to compare members based just on their name? Do we want to check inferred signatures as well?

Reply:
I think it is better to check for the inferred signatures too here.

Reply:
Could you add a test that would show a difference, if overloads are present?